src/Symfony/Component/Cache/Serializer/PhpSerializer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Serializer;
+
+use Symfony\Component\Cache\Exception\CacheException;
+use Symfony\Component\Cache\SerializerInterface;
+
+class PhpSerializer implements SerializerInterface
+{
+    public function serialize($data)
+    {
+        return serialize($data);
+    }
+
+    public function unserialize($serialized)
+    {
+        $unserializeCallbackHandler = ini_set('unserialize_callback_func', __CLASS__.'::handleUnserializeCallback');
+        try {
+            $value = unserialize($serialized);
+            if (false === $value && serialize(false) !== $serialized) {
```

Comment:
this is a performance critical code, the previous logic should be preserved (checking the serialized "false" before unserializing)

Reply:
It does unnecessary unserialize only on false value. Does storing false value as opcache entry is so frequent? But, ok. I would change it back.